### ISSUE TITLE
"Value cannot be null" error when AzureWebJobsStorage is missing for output binding

### ISSUE THREAD

**Issue by EricJizbaMSFT** [Azure/azure-functions-core-tools#1460]
```
func --version
2.7.1480
```

### Repro steps:
1. Create a new project (language shouldn't matter, but I did JS)
1. Add an http trigger
1. Add a Cosmos DB output binding to that trigger
1. Make sure extension bundle is set in host.json
1. Make sure AzureWebJobsStorage is _not_ set in local.settings.json

### Expected:
A readable error message saying `AzureWebJobsStorage` needs to be set. Honestly I would expect this to be the same error message that occurs if you have a Cosmos DB trigger (instead of output binding) without `AzureWebJobsStorage` set. That error is much better.

### Actual:
Error message:
```
Value cannot be null.
Parameter name: provider
```

_originally brought up by @askpatrickw in https://github.com/microsoft/vscode-azurefunctions/issues/1403#issuecomment-515238736_

**Comment by ahmelsayed** [comment]
Requiring storage for all output/input bindings would be a breaking change. Today it's only required if you have anything other than http or kafka **triggers**, but not bindings.

@fabiocav do you have a list of the input/output bindings that require `AzureWebJobsStorage`

**Comment by EricJizbaMSFT** [comment]
@ahmelsayed yeah I'm not asking you to change the behavior of when it's required - just to print out a better error message than "Value cannot be null."

**Comment by ahmelsayed** [comment]
I understand, and I agree that error messages should be better.

The problem is that this error message is coming from the runtime, not the core-tools. The core-tools sets things up, then invokes the runtime. It should be the runtime's responsibility to properly handle invalid function settings, but it doesn't. It just re-throws those exceptions. Since it was too confusing for users, and apparently too complicated for the runtime to display the right errors, the [core-tools parses function.json itself](https://github.com/Azure/azure-functions-core-tools/blob/f53563b622ee68f00811ddaaaa67d1199067f8ad/src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs#L366-L391) and maintains a list of what works with what
